fccodegen/data/tests/weblab_model.pyx

```diff
@@ -158,42 +156,43 @@ cdef class GeneratedModel_Proto_tmpHzG5bQ(CvodeSolver): # TODO Generate name
         self.freeVariableName = "time"
         self.freeVariable = 0.0
 
-        # TODO Generate this
-        self.state = np.zeros(2)
+        # State values
+        self.state = np.zeros(4)
+
+        # Mapping from cmeta ids to state indices
+        # TODO Is this correct? Not all states need to have a cmeta id!
```

Comment:
The docstring says "stateVarMap: a mapping from variable name to index within the state variable vector, for use in our ModelWrapperEnvironment". Variable name here is the name as known by the protocol, i.e. oxmeta name rather than cmeta:id. A state variable which isn't annotated therefore doesn't appear in this map and can't be altered (or read individually) from the protocol.

Reply:
Thanks!